ft-api/src/api.rs

```diff
@@ -130,13 +116,45 @@ where
         });
     }
 
-    resp.json().map_err(PageError::DeserializeError)
+    let status = resp.status();
+
+    let resp_value: Result<ApiResponse<serde_json::Value>, reqwest::Error> = resp.json();
+    match resp_value {
+        Ok(r) => {
+            if !r.success {
+                return Err(PageError::UnexpectedResponse {
+                    code: status,
+                    body: r.error.map_or("Something went wrong".to_string(), |x| {
+                        x.into_iter()
+                            .map(|(k, v)| k + ": " + &v)
+                            .collect::<Vec<_>>()
+                            .join("\n")
+                    }),
+                });
+            }
+            match r.result {
+                Some(v) => serde_json::from_value(v).map_err(PageError::SerdeDeserializeError),
+                None => {
+                    return Err(PageError::UnexpectedResponse {
+                        code: status,
+                        body: "Response is not present".to_string(),
+                    })
+                }
+            }
+        }
+        Err(err) => {
+            return Err(PageError::UnexpectedResponse {
+                code: status,
+                body: err.to_string(),
+            })
+        }
+    }
 }
 
-pub fn action<T, B>(url: &str, body: B, tid: Option<String>) -> crate::Result<ApiResponse<T>>
+pub fn action<T, B>(url: &str, body: B, tid: Option<String>) -> crate::Result<T>
 where
     T: serde::de::DeserializeOwned,
-    B: serde::Serialize,
+    B: Into<reqwest::blocking::Body> + serde::Serialize,
```

Comment:
@amitu Mostly fixes the compilation error

Reply:
When used this, it was sending correct data in the request body, or may be server was not able to decode the body 
```.body(reqwest::blocking::Body::from(serde_json::to_vec(&body)?))```

I was getting error ```error: NotFound { key: "data" }```

when sending it with body only so it worked fine.

Reply:
Because you are not sending a struct with a field named "data". This is what our API expects. I have no idea how is it getting fixed by `eqwest::blocking::Body::from()`, but it is wrong. You have to send the correct data.

Reply:
It is working fine now with `serde::to_string` only 
`.body(serde_json::to_string(&body)?)`